pkg/deployment/resources/pdbs.go

```diff
@@ -0,0 +1,105 @@
+package resources
+
+import (
+	"fmt"
+
+	api "github.com/arangodb/kube-arangodb/pkg/apis/deployment/v1alpha"
+	"github.com/arangodb/kube-arangodb/pkg/util/k8sutil"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func min(a int, b int) int {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+// EnsurePDBs ensures Pod Distruption Bugets for different server groups in Cluster mode
+func (r *Resources) EnsurePDBs() error {
+
+	// Only in Cluster and Production Mode
+	spec := r.context.GetSpec()
+	if spec.IsProduction() && spec.GetMode().IsCluster() {
+
+		minAgents := spec.GetServerGroupSpec(api.ServerGroupAgents).GetCount() - 1
+		minDBServers := spec.GetServerGroupSpec(api.ServerGroupDBServers).GetCount() - 1
+		minCoordinators := min(spec.GetServerGroupSpec(api.ServerGroupCoordinators).GetCount()-1, 2)
+
+		if err := r.ensurePDBForGroup(api.ServerGroupAgents, minAgents); err != nil {
+			return err
+		}
+		if err := r.ensurePDBForGroup(api.ServerGroupDBServers, minDBServers); err != nil {
+			return err
+		}
+		if err := r.ensurePDBForGroup(api.ServerGroupCoordinators, minCoordinators); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func pdbNameForGroup(depl string, group api.ServerGroup) string {
+	return fmt.Sprintf("%s-%s-pdb", depl, group.AsRole())
+}
+
+func newPDB(minAvail int, deplname string, group api.ServerGroup, owner metav1.OwnerReference) *policyv1beta1.PodDisruptionBudget {
+	return &policyv1beta1.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            pdbNameForGroup(deplname, group),
+			OwnerReferences: []metav1.OwnerReference{owner},
```

Comment:
What about namespace?

Reply:
Namespace is implicitly set due by the api-server, since the client-set already has the namespace.